test/src/sensors/drag-sensor.test.js

```diff
@@ -0,0 +1,477 @@
+        .toMatchObject(['dragend', sensor._onDragEnd, false]);
+
+      expect(container.addEventListener.getCall(4).args)
+        .toMatchObject(['drop', sensor._onDragDrop, false]);
+
+      expect(document.addEventListener.getCall(0).args)
+        .toMatchObject(['mouseup', sensor._onMouseUp, true]);
+    });
+  });
+
+  describe('#detach', () => {
+    let container = null;
+
+    beforeEach(() => {
+      container = document.createElement('div');
+
+      sinon.spy(container, 'removeEventListener');
+      sinon.spy(document, 'removeEventListener');
+    });
+
+    afterEach(() => {
+      container.removeEventListener.restore();
+      document.removeEventListener.restore();
+
+      container = null;
+    });
+
+    test('should remove event listeners', () => {
+      const sensor = new DragSensor([container]);
+
+      sensor.detach();
+
+      expect(container.removeEventListener.callCount).toBe(5);
+
+      expect(container.removeEventListener.getCall(0).args)
+        .toMatchObject(['mousedown', sensor._onMouseDown, true]);
+
+      expect(container.removeEventListener.getCall(1).args)
+        .toMatchObject(['dragstart', sensor._onDragStart, false]);
+
+      expect(container.removeEventListener.getCall(2).args)
+        .toMatchObject(['dragover', sensor._onDragOver, false]);
+
+      expect(container.removeEventListener.getCall(3).args)
+        .toMatchObject(['dragend', sensor._onDragEnd, false]);
+
+      expect(container.removeEventListener.getCall(4).args)
+        .toMatchObject(['drop', sensor._onDragDrop, false]);
+
+      expect(document.removeEventListener.getCall(0).args)
+        .toMatchObject(['mouseup', sensor._onMouseUp, true]);
+    });
+  });
+
+  describe('#_onDragStart', () => {
+    let sandbox;
+    let dragSensor;
+
+    beforeEach(() => {
+      listenToSensorEvents();
+      sandbox = createSandbox(sampleMarkup);
+      const container = sandbox.querySelectorAll('ul');
+      dragSensor = new DragSensor(container, {
+        delay: 0,
+        type: 'expected type',
+      });
+      dragSensor.attach();
+    });
+
+    afterEach(() => {
+      restoreSensorEvents();
+      sandbox.parentNode.removeChild(sandbox);
+    });
+
+    test('should trigger `drag:start` sensor event on dragstart', () => {
+      const draggable = sandbox.querySelector('li');
+      document.elementFromPoint = () => draggable;
+
+      // Fire primary dragstart event
+      triggerEvent(draggable, 'dragstart', {
+        dataTransfer: getDataTransferStub(),
+      });
+
+      expect(getSensorEventsByType('drag:start').length)
+        .toBe(1);
+    });
+
+    test('should configure DragStartSensorEvent correctly', () => {
+      const draggable = sandbox.querySelector('li');
+      const dataTransferStub = getDataTransferStub();
+      document.elementFromPoint = () => draggable;
+
+      // Fire primary dragstart event
+      const dispatchedBrowserEvent = triggerEvent(draggable, 'dragstart', {
+        clientX: 37,
```

Comment:
Put these test numbers in a variable so if they are every changed you don't have to update your expect statements

Reply:
Same suggestion for all numbers below.

Reply:
@lodenrogue All set, thanks!